Google.PowerShell/Compute/GceTargetPoolCmdlets.cs

```diff
@@ -0,0 +1,323 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// Licensed under the Apache License Version 2.0.
+
+using Google.Apis.Compute.v1;
+using Google.Apis.Compute.v1.Data;
+using Google.PowerShell.Common;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Google.PowerShell.ComputeEngine
+{
+    /// <summary>
+    /// <para type="synopsis">
+    /// Gets Google Compute Engine target pools.
+    /// </para>
+    /// <para type="description">
+    /// Lists target pools of a project, or gets a specific one.
+    /// </para>
+    /// <example>
+    /// <code>PS C:\> Get-GceTargetPool</code>
+    /// <para>Lists all target pools for the default project.</para>
+    /// </example>
+    /// <example>
+    /// <code>PS C:\> Get-GceTargetPool -Region us-central1</code>
+    /// <para>Lists all target pools in region "us-central1" for the default project.</para>
+    /// </example>
+    /// <example>
+    /// <code>PS C:\> Get-GceTargetPool "my-target-pool"</code>
+    /// <para>Gets the target pool named "my-target-pool" in the default project and zone</para>
+    /// </example>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "GceTargetPool", DefaultParameterSetName = ParameterSetNames.OfProject)]
+    [OutputType(typeof(TargetPool))]
+    public class GetGceTargetPoolCmdlet : GceCmdlet
+    {
+        private class ParameterSetNames
+        {
+            public const string OfProject = "OfProject";
+            public const string OfRegion = "OfRegion";
+            public const string ByName = "ByName";
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// The project the target pools belong to. Defaults to the project in the Cloud SDK config.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.OfProject)]
+        [Parameter(ParameterSetName = ParameterSetNames.OfRegion)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByName)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The region of the forwaring rule to get. Defaults to the region in the Cloud SDK config.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.OfRegion, Mandatory = true)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByName)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Region)]
+        public string Region { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The name of the target pool to get.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByName, Mandatory = true, Position = 0)]
+        public string Name { get; set; }
+
+        // TODO(jimwp) Understand what this health status is an uncomment this parameter.
```

Comment:
Colon after `)`. Also, can you elaborate? "what this health status is an"? What do you need to do to resolve this TODO?

Also, don't leave commented out code (//[Parameter]). Just have the TODO say, "figure out if this should be a Parameter" or something.

And finally, what is the meaning of having a `SwitchParameter` type that is not actually used? Should we just delete the public property as well?